x/ownership/internal/keeper/keeper.go

```diff
@@ -77,12 +82,29 @@ func (k Keeper) Delete(ctx sdk.Context, owner sdk.AccAddress, resourceHash hash.
 		return fmt.Errorf("resource %q do not have any ownership", resourceHash)
 	}
 
+	// transfer all spendable coins from resource address to owner
+	addr := sdk.AccAddress(crypto.AddressHash(resourceHash))
+	coins := k.bankKeeper.GetCoins(ctx, addr)
+	if err := k.bankKeeper.SendCoins(ctx, addr, owner, coins); err != nil {
```

Comment:
Can we do a safe test here, it's not really clear that `SendCoins` will manage correctly if we send empty coin and this might change in the future.

Can we do something like
```go
if coins != 0 {
  if err := k.bankKeeper.SendCoins(ctx, addr, owner, coins); err != nil {
    return err
  }
}
```

At least we will know that we don't send if it's 0, also, in that case, we will not emit an event with empty value.